openedx/features/edly/tests/test_middleware.py

```diff
@@ -71,3 +75,137 @@ def test_super_user_has_all_sites_access(self):
 
         response = client.get('/', follow=True)
         assert response.status_code == 200
+
+
+class SettingsOverrideMiddlewareTests(TestCase):
+    """
+    Tests settings override middleware for sites.
+    """
+    def setUp(self):
+        """
+        Create environment for settings override middleware tests.
+        """
+        self.user = EdlyUserFactory()
+        self.request = RequestFactory().get('/')
+        self.request.user = self.user
+        self.request.site = SiteFactory()
+        EdlySubOrganizationFactory(lms_site=self.request.site)
+
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.cookies.load(
+            {
+                settings.EDLY_USER_INFO_COOKIE_NAME: cookies._get_edly_user_info_cookie_string(self.request)
+            }
+        )
+        self.client.login(username=self.user.username, password='test')
+        self.default_settings = {key: getattr(settings, key, None) for key in [
+                'SITE_NAME', 'CMS_BASE', 'LMS_BASE', 'LMS_ROOT_URL', 'SESSION_COOKIE_DOMAIN',
+                'LOGIN_REDIRECT_WHITELIST', 'DEFAULT_FEEDBACK_EMAIL', 'CREDENTIALS_PUBLIC_SERVICE_URL'
+            ]
+        }
+
+    def _assert_settings_equal(self, expected_settings_values):
```

Comment:
```python
def _assert_response_settings_values(self, response, expected_site_configuration_values):
    """
    Validate configuration settings in HTTP response.
    """
    for config_key, expected_config_value in expected_site_configuration_values.items():
        assert expected_config_value == getattr(response.context['settings'], config_key)
```

Reply:
these settings are django settings not available in response. that's why i set this name